### PR TITLE
Update translation hashing

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Generate only a single hash for each language and use that hash on each partial translation file. This will significantly reduce the translation metadata that we need to include in the frontend.

Also realized the build ran certain tasks multiple times. This is no longer the case and it went from 48s on my machine to 38s.

In dev mode we will now use a static hash, so that updating translation files will no longer generate new hashes.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "fecha": "^3.0.2",
     "fuse.js": "^3.4.4",
     "google-timezones-json": "^1.0.2",
+    "gulp-rename": "^2.0.0",
```
Comment (bramkragten) [inline package.json]: Is a dev dependency?

**Review by bramkragten — COMMENTED** [review]
I think I miss some files?

**Comment by balloob** [comment]
The diff for translations.js is minimized because so many changes

File: build-scripts/gulp/translations.js [hunk 2]
```diff
@@ -172,235 +164,215 @@ tasks.push(taskName);
...
-        })
-      )
-      .pipe(gulp.dest(workDir));
-  })
+  "build-translation-fingerprints",
+  function fingerprintTranslationFiles() {
+    // Fingerprint full file of each language
+    const files = fs.readdirSync(fullDir);
+    for (let i = 0; i < files.length; i++) {
+      fingerprints[files[i].split(".")[0]] = {
```
Comment (balloob) [inline build-scripts/gulp/translations.js]: Here is where the magic is happening. We now just hash the full language files and then use those for all files belonging to that language

**Review by bramkragten — APPROVED** [review]

**Pull request merged**